cabal-install/Distribution/Client/Config.hs

```diff
@@ -445,7 +445,7 @@ initialSavedConfig = do
   return mempty {
     savedGlobalFlags     = mempty {
       globalCacheDir     = toFlag cacheDir,
-      globalRemoteRepos  = toNubList [addInfoForKnownRepos defaultRemoteRepo],
+      globalRemoteRepos  = toNubList [defaultRemoteRepo],
```

Comment:
This wasn't adding anything, and for the secure stuff we don't actually want to persist these things as it'd make switching from opt-in to opt-out impossible.